scripts/count-bugs.py

```diff
@@ -3,25 +3,40 @@
 # Copyright (c) 2019 Patryk Obara <patryk.obara@gmail.com>
 # SPDX-License-Identifier: GPL-2.0-or-later
 
-# This script prints a summary snippet of information out of reports created
-# by scan-build or analyze-build for Clang's static code analysis.
-#
-# Usage: ./count-warnings.py path/to/report/index.html
-#
+"""
+This script prints a summary snippet of information out of reports created
+by scan-build or analyze-build for Clang's static code analysis.
+
+This script counts the number of reported issues and compares that
+against provided maximum. if the count exceeds the maximum then the
+script will return a status of 1 (failure), otherwise the script
+returns 0 (success).
+
+If you prefer to not set a maximum, then supply a negative maximum
+such as -1.  If a maximum is not provided on the command line then
+the internally-defined maximum is used.
```

Comment:
Will update.

Reply:
Yeah - my first edit switched to a pure command line argument, which made sense when you have the maximums hard-coded in the YAML file.. but then I thought that you might have been using the hardcoded values for programming at home, because otherwise you'd have to remember whatever the previous number was.  So I added that back.

Now the script is even slicker.. I really like your recommendation and how it worked out.  Give it a go with all the cases on the command line; definitely nice without much code!

Fixed in bfc26ba3